src/modules/AutoinstClone.rb

```diff
@@ -98,88 +93,49 @@ def CommonClone(_resource, resourceMap)
     end
 
     # Create a list of clonable resources
-    # @return [Array] list to be used in widgets
+    #
+    # @return [Array<Yast::Term>] list to be used in widgets
     def createClonableList
-      items = []
-      Builtins.foreach(Y2ModuleConfig.ModuleMap) do |def_resource, resourceMap|
-        Builtins.y2debug(
-          "r: %1 => %2",
-          def_resource,
-          Ops.get_string(resourceMap, "X-SuSE-YaST-AutoInstClonable", "false")
-        )
-        clonable = Ops.get_string(
-          resourceMap,
-          "X-SuSE-YaST-AutoInstClonable",
-          "false"
-        ) == "true"
-        if clonable || "partitioning" == def_resource ||
-            "software" == def_resource || # has no desktop file
-            "bootloader" == def_resource # has no desktop file
-          desktop_file = Builtins.substring(
-            Ops.get_string(resourceMap, "X-SuSE-DocTeamID", ""),
-            4
-          )
-          name = Builtins.dpgettext(
-            "desktop_translations",
-            "/usr/share/locale/",
-            Ops.add(
-              Ops.add(Ops.add("Name(", desktop_file), ".desktop): "),
-              Ops.get_string(resourceMap, "Name", "")
-            )
-          )
-          if name ==
-              Ops.add(
-                Ops.add(Ops.add("Name(", desktop_file), ".desktop): "),
-                Ops.get_string(resourceMap, "Name", "")
-              )
-            name = Ops.get_string(resourceMap, "Name", "")
-          end
-          # Set resource name, if not using default value
-          resource = Ops.get_string(
-            resourceMap,
-            "X-SuSE-YaST-AutoInstResource",
-            ""
-          )
-          resource = def_resource if resource == ""
-          items = if resource != ""
-            Builtins.add(items, Item(Id(resource), name))
-          else
-            Builtins.add(items, Item(Id(def_resource), name))
-          end
-        end
-      end
-      # sort items for nicer display
-      items = Builtins.sort(
-        Convert.convert(items, from: "list", to: "list <term>")
-      ) do |x, y|
-        Ops.less_than(Ops.get_string(x, 1, "x"), Ops.get_string(y, 1, "y"))
+      module_map = Y2ModuleConfig.ModuleMap
+      clonable_items = module_map.each_with_object([]) do |(def_resource, resource_map), items|
+        log.debug "r: #{def_resource} => #{resource_map["X-SuSE-YaST-AutoInstClonable"]}"
+        clonable = resource_map["X-SuSE-YaST-AutoInstClonable"] == "true"
+        next unless clonable
+
+        desktop_file = resource_map.fetch("X-SuSE-DocTeamID", "")[4..]
```

Comment:
ugh, it is first time I see this. Can you please document it in https://github.com/yast/yast-yast2/blob/master/doc/desktop_file.md

Reply:
Done: https://github.com/yast/yast-yast2/pull/1052.